File: KadaneAlgorithm/kadane.lua
function kadane(arr)
	local max_sum = -math.huge
	local current_sum = 0
	local n = #arr

	for i = 1, n do
		current_sum = current_sum  + arr[i]

		if max_sum < current_sum then max_sum = current_sum end
		if current_sum < 0 then current_sum = 0 end
	end

	return max_sum
end

function main()
	io.write("Enter number of elements: ")
	local n = io.read("*n")
	local arr = {}
	
	for i = 1, n do
		io.write("Enter an element: ")
		arr[i] = io.read("*n")
	end

	print("The largest sub array sum is: "..kadane(arr))
end

main()

File: KadaneAlgorithm/kadane.cpp
#include <iostream>

#define cin std::cin
#define cout std::cout

int kadane(int arr[], int n) {
	int max_sum = -99999999;
	int current_sum = 0;

	for(int i = 0; i < n; i++) {
		current_sum = current_sum + arr[i];

		if(max_sum < current_sum) {
			max_sum = current_sum;
		}
		if(current_sum < 0) {
			current_sum = 0;
		}
	}

	return max_sum;
}

int main() {
	int n;

	cout << "Enter number of elements: ";
	cin >> n;

	int arr[n];

	for(int i = 0; i < n; i++) {
		cout << "Enter an element: ";
		cin >> arr[i];
	}

	cout << "The largest sub array sum is: " << kadane(arr, n);
}
